spec/lib/hubspot/company_properties_spec.rb

```diff
@@ -14,158 +14,325 @@
-
-        it 'should raise an error' do
-          expect { Hubspot::CompanyProperties.delete!(name) }.to raise_error(Hubspot::RequestError)
+      context "when the company property does not exist" do
+        it "raises an error" do
+          VCR.use_cassette("company_properties/delete_non_property") do
+            expect {
+              Hubspot::CompanyProperties.delete!("non-existent")
+            }.to raise_error(Hubspot::RequestError)
+          end
         end
       end
     end
   end
 
-  describe 'Groups' do
-    describe '.groups' do
-      context 'with no filter' do
-        cassette 'company_properties/all_groups'
+  describe "Groups" do
+    describe ".groups" do
+      it "returns all groups" do
+        VCR.use_cassette("company_properties/all_groups") do
+          group_name = "group_awesome"
+          Hubspot::CompanyProperties.create_group!(name: group_name)
 
-        it 'should return all groups' do
-          expect(Hubspot::CompanyProperties.groups).to eql(example_groups)
-        end
-      end
+          response = Hubspot::CompanyProperties.groups
 
-      let(:groups) { %w(calltrackinginfo emailinformation) }
+          assert_hubspot_api_request(:get, "/companies/v2/groups")
 
-      context 'with included groups' do
-        cassette 'company_properties/groups_included'
+          group_names = response.map { |group| group["name"] }
+          expect(group_names).to include(group_name)
 
-        it 'should return the specified groups' do
-          response = Hubspot::CompanyProperties.groups({}, { include: groups })
-          response.each { |p| expect(groups.include?(p['name'])).to be true }
+          Hubspot::CompanyProperties.delete_group!(group_name)
         end
       end
 
-      context 'with excluded groups' do
-        cassette 'company_properties/groups_not_excluded'
+      context "when included groups are provided" do
+        it "returns the specified groups" do
+          VCR.use_cassette("company_properties/groups_included") do
+            group_a = "socialmediainformation"
+            group_b = "companyinformation"
+
+            Hubspot::CompanyProperties.create!({
+              name: "instagram_handle",
+              label: "Instagram Handle",
+              description: "The company's Instagram handle.",
+              groupName: group_a,
+              type: "string",
+              fieldType: "text"
+            })
+            Hubspot::CompanyProperties.create!({
+              name: "fax_number",
+              label: "Fax Number",
+              description: "The company fax number.",
+              groupName: group_b,
+              type: "string",
+              fieldType: "text"
+            })
+
+            response = Hubspot::CompanyProperties.groups(
+              {},
+              { include: group_a }
+            )
+
+            assert_hubspot_api_request(:get, "/companies/v2/groups")
+
+            Hubspot::CompanyProperties.delete!("instagram_handle")
```

Comment:
Same question here around ensuring that these happen even if the exercise or expectation part of this test fails.